eppo_client/client.py

```diff
@@ -36,143 +35,146 @@ def __init__(
-            )
-        except ValueError as e:
-            # allow ValueError to bubble up as it is a validation error
-            raise e
-        except Exception as e:
-            if self.__is_graceful_mode:
-                logger.error("[Eppo SDK] Error getting assignment: " + str(e))
-                return None
-            raise e
+        variation_jsons = self.get_assignment_variation(
+            subject_key,
+            flag_key,
+            subject_attributes,
+            VariationType.JSON,
+            default=default,
+        )
+        if variation_jsons is None:
+            return None
+        return json.loads(variation_jsons)
 
     @deprecated(
         "get_assignment is deprecated in favor of the typed get_<type>_assignment methods"
     )
     def get_assignment(
-        self, subject_key: str, flag_key: str, subject_attributes=dict()
+        self,
+        subject_key: str,
+        flag_key: str,
+        subject_attributes: Optional[SubjectAttributes] = None,
+        default=None,
     ) -> Optional[str]:
+        return self.get_assignment_variation(
+            subject_key, flag_key, subject_attributes, default=default
+        )
+
+    def get_assignment_variation(
+        self,
+        subject_key: str,
+        flag_key: str,
+        subject_attributes: Optional[SubjectAttributes] = None,
+        expected_variation_type: Optional[VariationType] = None,
+        default=None,
+    ):
         try:
-            assigned_variation = self.get_assignment_variation(
-                subject_key, flag_key, subject_attributes
-            )
-            return (
-                assigned_variation.value
-                if assigned_variation is not None
-                else assigned_variation
+            result = self.get_assignment_detail(
+                subject_key, flag_key, subject_attributes, expected_variation_type
             )
+            if not result or not result.variation:
+                return default
+            return result.variation.value
         except ValueError as e:
             # allow ValueError to bubble up as it is a validation error
             raise e
         except Exception as e:
             if self.__is_graceful_mode:
                 logger.error("[Eppo SDK] Error getting assignment: " + str(e))
-                return None
+                return default
             raise e
 
-    def get_assignment_variation(
+    def get_assignment_detail(
         self,
         subject_key: str,
         flag_key: str,
-        subject_attributes: Any,
-        expected_variation_type: Optional[str] = None,
-    ) -> Optional[VariationDto]:
+        subject_attributes: Optional[SubjectAttributes] = None,
+        expected_variation_type: Optional[VariationType] = None,
+    ) -> Optional[FlagEvaluation]:
         """Maps a subject to a variation for a given experiment
         Returns None if the subject is not part of the experiment sample.
 
```

Comment:
Doc string needs to be updated. References to `experiment` should be removed or replaced

Reply:
Good catch!